aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/AWSXRayRecorder.java

```diff
@@ -1029,4 +1054,24 @@ public String currentFormattedId() {
             return null;
         }
     }
+
+    /**
+     * Internal-only utility method.
+     */
+    @Deprecated
+    public final <T> T getRandomBytes(int bytes, Function<byte[], T> mapper) {
```

Comment:
I know @rhernandez35 you want to avoid exposing these, but I'm not sure how I feel about adding new internal-only methods to an already very bloated Recorder class. IMO these methods can be extracted into a public interface/class (like `RandomGenerator`) and exposed to customers. That is to some degree what we're requesting OpenTelemetry, a pluggable ID generation experience. I'm not sure if customers would bother using it unless they really knew what they were doing, and we can still guarantee randomness with our *own* generators.

@anuraaga I'd appreciate you weighing in on this too, I could be completely off the mark here.

Reply:
If you're fine with increasing the surface area then I'll gladly do it since the code would be so much cleaner. However, my goal was to avoid providing customers a way to evade our curated selection of randomness providers. I really don't want someone to think "oh hey I can save memory by just using `Math#random`" and accidentally add a global synchronization point on every new ID creation.

Adding these methods here was definitely a gigantic hack, but it seemed like the lesser of two evils.

Reply:
I lean towards having only two types of random - OpenTelemetry supports trace ID customization since it needs to support all backends, including weird ones like X-Ray. X-Ray SDK on the flip side can't be used with any other backend so we can just limit the options.

Instead of the pattern of deprecated method for internal API, we have the `.internal` package - can we move he internal pieces there?

Reply:
sure, I'm fine with that if you are.

I can say from experience though that users don't really care what the fully qualified name for a class is when they use it. If it's pubic and their IDE can autoresolve it, they often don't even know what package it's in. If it's public, it _will_ be used in ways you can't anticipate.

Reply:
I think the `internal` package sounds good to me. Sure customers *can* override it if they want, but it will not be a publicly documented configuration like the SDK's other configs (as a matter of fact the JavaDoc should specifically call out that it's an internal-only API that is not meant to be used by consumers of the SDK, which may help guide some of those IDE users 😉 ).